easyrelease-lib/src/main/groovy/eu/inloop/easyrelease/plugin/Util.groovy

```diff
@@ -0,0 +1,142 @@
+            }
+        }
+    }
+
+    static def getNameString(project) {
+        def projectDir = project.projectDir
+        def propFileName = 'easyrelease.properties'
+        def propFile = new File("$projectDir/$propFileName")
+        if (propFile.canRead()) {
+            def props = new Properties()
+            props.load(new FileInputStream(propFile))
+
+            if (props["APK_NAME"] != null) {
+                return props["APK_NAME"]
+            }
+            println "$TAG APK_NAME is null, using the default one "
+        }
+        return DEFAULT_APK_STRUCTURE
+    }
+
+    static def getBinding(project, variant) {
+        def now = Calendar.getInstance()
+        def binding = [
+                projectName : project.name,
+                variantName : variant.name,
+                // first check version code and name from Gradle build script, then from AndroidManifest.xml
+                variantVersionCode : variant.versionCode ? variant.versionCode : getVersionCode(project),
+                variantVersionName : variant.versionName ? variant.versionName : getVersionName(project),
+                dateYear : now.get(Calendar.YEAR),
+                dateMonth : now.get(Calendar.MONTH) + 1,
+                dateDay : now.get(Calendar.DAY_OF_MONTH),
+                dateHour : now.get(Calendar.HOUR_OF_DAY),
+                dateMinute : now.get(Calendar.MINUTE),
+                dateSecond : now.get(Calendar.SECOND),
+                dateMillisecond : now.get(Calendar.MILLISECOND)
+        ]
+
+        try {
+            def repo = Grgit.open(project.file('..'))
+            // buildNumber based on branch commit count might crash when git path is
+            // set incorrectly, leaving out until further research
+            // binding.buildNumber = 'git rev-list --count HEAD'.execute().text.trim()
+            binding.headShortHash = repo.head().abbreviatedId
+            binding.headAuthorName = repo.head().author.getName()
+            binding.headAuthorEmail = repo.head().author.getEmail()
+            binding.headCommiterName = repo.head().committer.getName()
+            binding.headCommiterEmail = repo.head().committer.getEmail()
+            def branchFullName = repo.branch.current.name.split("//")
+            binding.branchName = branchFullName[branchFullName.length-1]
+            binding.each {
+                println (it)
+            }
+            return binding
+        } catch (all) {
+            println "Exception raised: " + all
+        }
+
+        return binding
+    }
+
+    /**
+     * Parsing AndroidManifest.xml to return versionName
+     */
+    static def getVersionName(project) {
+        def androidManifestPath = project.android.sourceSets.main.manifest.srcFile
+        def manifestText = project.file(androidManifestPath).getText()
+        def patternVersionNumber = java.util.regex.Pattern.compile("versionName=\"(\\d+)\\.(\\d+)\\.(\\d+)\"")
```

Comment:
This decomposition and again composition seems useless and limits the version name to be x.y.z